### PR TITLE
Port/fix/deleting items

### PR DESCRIPTION
**Pull request by Raycoms**
Port the fixes pr to 1.10:

Fixes builder behavior
Fixes items going missing in the builder GUI
Fixes problems with rendering the colony border
Fixes workers deleting items when picking up with full inventory
(+Some more)

Review please

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 67 issues
* ![MAJOR][MAJOR] 22 major
* ![MINOR][MINOR] 42 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.321 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 56 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): class "EntityCitizen" has 95 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L359](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L359): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L360](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L360): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L360](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L360): Assign this magic number 0.02F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L361](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L361): Assign this magic number 0.6D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L362](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L362): Assign this magic number 6.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L391](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L391): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L602](https://github.com/Minecolonies/minecolonies/blob/faf78cdb2021693e7b1e181f6594196d1c7f638a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L602): Refactor this method to reduce its Cognitive Complexity from 25 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java [hunk 1]
```diff
@@ -200,6 +203,17 @@ public void serializeToView(@NotNull ByteBuf buf)
         }
     }
 
+    @Override
+    public void registerBlockPosition(@NotNull final Block block, @NotNull final BlockPos pos, @NotNull final World world)
+    {
+        //Only the chests because he shouldn't fill up the furnaces.
+        if (block instanceof BlockChest)
```
Comment (Asherslab) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java:210]: Wouldn't this only do BlockChests? i made one that allowed TileEntities?

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java:210]: No, this is only for blocks which should register with the builderHut itself.

**Comment by Asherslab** [comment]
Ahhh, lol. I thought it was a different class.

**Comment by Asherslab** [comment]
:shipit:

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1372/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Raycoms** [comment]
@OrionDevelopment

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 67 issues
* ![MAJOR][MAJOR] 22 major
* ![MINOR][MINOR] 42 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.321 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 56 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): class "EntityCitizen" has 95 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L359](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L359): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L360](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L360): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L360](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L360): Assign this magic number 0.02F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L361](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L361): Assign this magic number 0.6D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L362](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L362): Assign this magic number 6.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L391](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L391): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L602](https://github.com/Minecolonies/minecolonies/blob/9232d546ca1ce5c3305fb6c357491af1279e88c1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L602): Refactor this method to reduce its Cognitive Complexity from 25 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**